src/SDK/Trace/SpanProcessor/BatchSpanProcessor.php

```diff
@@ -72,6 +72,7 @@ public function __construct(
      */
     public function onStart(ReadWriteSpanInterface $span, ?Context $parentContext = null): void
     {
+        $span->addResourceToAttributes();
```

Comment:
I think that the span converters take care of this (a couple that I checked, in `src/contrib` do, anyway). So it looks to me like you just need to make sure that the detected resources (via `Composite` detector?) are added to the tracer provider, and as far as I can see they're passed on to a Tracer, then SpanBuilder, and finally to a Span.

Reply:
Detected resources are already present in tracer provider, but only as ResourceInfo. When we export on span end, we report only attributes of span, not the resource info. With this function addResourceToAttributes, I am adding attributes present in ResourceInfo to span attributes and calling it in onStart because ResourceInfo will be initialised with tracer.

Reply:
hmm, I thought that is what this line was doing? https://github.com/open-telemetry/opentelemetry-php/blob/main/src/Contrib/Otlp/SpanConverter.php#L176 - iterating over a span's resources (`ResourceInfo`), and adding them as attributes to a span.

Reply:
Yes, you are correct. This line does exactly that. But apart from Otlp SpanConverter, does this.

Reply:
It seems like different exporters have different requirements for how resources are represented, eg https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#resource - if that's the case, it probably needs to be managed in the individual span converters in a similar way to how OTLP does it.